src/Netflix/Netflix.ts

```
@@ -0,0 +1,760 @@
+// lipsurf-plugins/src/Netflix/Netflix.ts
+/// <reference types="lipsurf-types/extension"/>
+declare const PluginBase: IPluginBase;
+
+// Messaging
+
+const LIPSURF_BOOT_SCRIPT_ID = "lipsurf-netflix-script";
+const TO_PAGE_PROOF_KEY = String(Math.floor(Math.random() * 1000));
+const FROM_PAGE_PROOF_KEY = String(Math.floor(Math.random() * 1000));
+
+type Message<T> = { proofKey: string; payload: T };
+
+const sendMessage = (payload: Command) =>
+  (window as any).postMessage(
+    JSON.stringify({ proofKey: TO_PAGE_PROOF_KEY, payload } as Message<Command>)
+  );
+
+/**
+ * Receive Message
+ */
+(() => {
+  window.addEventListener("message", ev => receiveMessage(ev.data));
+  const receiveMessage = (messageStr: string) => {
+    try {
+      const message = JSON.parse(messageStr) as Message<Command>;
+      if (message.proofKey === FROM_PAGE_PROOF_KEY) {
+        const command = message.payload;
+        switch (command.key) {
+          case "changeText":
+            return handleChangeTextAnswer(command);
+          case "changeAudio":
+            return handleChangeAudioAnswer(command);
+        }
+      }
+    } catch {
```

Better to not catch in that case if anything to let the error propagate.

---

I notice you catch errors in many places below. Sometimes it makes more sense to let the error propagate up. Right now, the page will catch pageFn errors and log them so we know which commands are failing where. But also we will likely give some sort of user feedback when a command fails (eg. make the live transcript turn red) in a subsequent version.

---

I actually intend to catch only the `JSON.parse` part. Not the other.
I think I'll make that clearer. The usage will be something like this.

```
consumeCommand(maybeCommand, proofKey, (command) => {
  ...
})